vm/src/error.rs

```diff
@@ -3,17 +3,37 @@ use thiserror::Error;
 /// Errors related to VM initialization and execution
 #[derive(Debug, Error)]
 pub enum NexusVMError {
+    /// not enough bytes available to complete instruction parse
+    #[error("partial instruction at pc:{0:x}")]
+    PartialInstruction(u32),
+
+    /// Invalid instruction size found during parse
+    #[error("invalid instruction size, {1}, at pc:{0:x}")]
+    InvalidSize(u32, u32),
+
     /// Invalid instruction format, could not parse
-    #[error("invalid instruction {1} at {0}")]
-    InvalidInstruction(u64, u32),
+    #[error("invalid instruction {1:x} at pc:{0:x}")]
+    InvalidInstruction(u32, u32),
 
     /// Unknown ECALL number
-    #[error("unknown syscall {1} at {0}")]
-    UnknownSyscall(u32, u32),
+    #[error("unknown ecall {1} at pc:{0:x}")]
+    UnknownECall(u32, u32),
 
-    /// Invalid memory address
-    #[error("invalid memory access {0:x}")]
-    SegFault(u32),
+    /// An error occurred reading file system
+    #[error(transparent)]
+    IOError(#[from] std::io::Error),
```

Comment:
nit: I realize this wasn't actually a change, but the comment seems overly prescriptive and susceptible to going stale (if it hasn't already).

```suggestion
    /// An I/O error occurred
    #[error(transparent)]
    IOError(#[from] std::io::Error),
```